src/cbc_sdk/workload/vulnerability_assessment.py

```diff
@@ -0,0 +1,902 @@
+        """
+        Performs the query and returns the results of the query in an iterable fashion.
+
+        Args:
+            from_row (int): The row to start the query at (default 0).
+            max_rows (int): The maximum number of rows to be returned (default -1, meaning "all").
+
+        Returns:
+            Iterable: The iterated query.
+        """
+        url = self._build_url("/_search")
+        current = from_row
+        numrows = 0
+        still_querying = True
+        while still_querying:
+            request = self._build_request(current, max_rows)
+            resp = self._cb.post_object(url, body=request)
+            result = resp.json()
+
+            self._total_results = result["num_found"]
+            self._count_valid = True
+
+            results = result.get("results", [])
+            for item in results:
+                yield item
+                current += 1
+                numrows += 1
+
+                if max_rows > 0 and numrows == max_rows:
+                    still_querying = False
+                    break
+
+            if current >= self._total_results:
+                break
+
+    def _run_async_query(self, context):
+        """
+        Executed in the background to run an asynchronous query.
+
+        Args:
+            context (object): Not used, always None.
+
+        Returns:
+            list: Result of the async query, which is then returned by the future.
+        """
+        url = self._build_url("/_search")
+        request = self._build_request(0, -1)
+        resp = self._cb.post_object(url, body=request)
+        result = resp.json()
+        self._total_results = result["num_found"]
+        self._count_valid = True
+        results = result.get("results", [])
+        return [item for item in results]
+
+    def sort_by(self, key, direction="ASC"):
+        """
+        Sets the sorting behavior on a query's results.
+
+        Example:
+            >>> cb.select(Vulnerabiltiy).sort_by("status")
+
+        Args:
+            key (str): The key in the schema to sort by.
+            direction (str): The sort order, either "ASC" or "DESC".
+
+        Returns:
+            VulnerabilityQuery: This instance.
+
+        Raises:
+            ApiError: If an invalid direction value is passed.
+        """
+        if direction not in VulnerabilityQuery.VALID_DIRECTIONS:
+            raise ApiError("invalid sort direction specified")
+        self._sortcriteria = {"field": key, "order": direction}
+        return self
+
+
+class AffectedAssetQuery(VulnerabilityQuery):
+    """Query Class for the Vulnerability"""
+
+    def __init__(self, doc_class, cb):
+        """
+        Initialize the VulnerabilityQuery.
+
+        Args:
+            doc_class (class): The model class that will be returned by this query.
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+        """
+        self._doc_class = doc_class
```

Comment:
This should be `self.vulnerability` instead of doc_class and `self.doc_class = Device`  see the other comment in DeviceVulnerability. Something that might be worth while to do is to gather all of the device_ids returned in the response and make a Device query like `self._cb.select(Device).set_device_ids([ids])` to return all the device models instead of just returning a list of devices